eth/beacon/types/crosslink_records.py

```diff
@@ -14,21 +14,17 @@ class CrosslinkRecord(rlp.Serializable):
     Note: using RLP until we have standardized serialization format.
     """
     fields = [
-        # What dynasty the crosslink was submitted in
-        ('dynasty', int64),
-        # slot during which crosslink was added
-        ('slot', int64),
-        # The block hash
-        ('hash', hash32),
+        # Slot during which crosslink was added
+        ('slot', uint64),
+        # Shard chain block hash
+        ('shard_block_hash', hash32),
     ]
 
     def __init__(self,
-                 dynasty: int,
                  slot: int,
-                 hash: Hash32) -> None:
+                 shard_block_hash: Hash32) -> None:
```

Comment:
I'm curious why this method was implemented as it appears to just be a passthrough.  Is there something about the base `__init__` that makes this necessary?

Reply:
I think yes.

Reply:
side note.  It'd be really cool if we could re-work `rlp.Serializable` so that it could be defined the same way that `NamedTuple` is defined so that the fields come with both types and sedes.

```python
class Header(rlp.Serializable):
    block_number: big_endian_int(32)
    ...
```